src/workerd/util/sqlite.c++

```diff
@@ -375,6 +542,33 @@ void SqliteDatabase::setupAuthorizer() {
       return SQLITE_DENY;
     }
   }, this));
+
+  // 4. Set a progress handler or use interrupt() to limit CPU time.
+  // TODO(sqlite): Call sqlite3_interrupt() at the same time as v8::Isolate::TerminateExecution().
+
+  // 5. Limit heap size.
+  // Annoyingly, this sets a process-wide limit. We'll set 128MB "soft" limit (to try to control
+  // how much page caching SQLite does) and 512MB "hard" limit (to block DoS attacks from taking
+  // down the whole system).
+  // TODO(perf): Revisit as popularity grows. Maybe make configurable? Maybe patch SQLite to allow
+  //   these to be controlled per-database? Is page caching even all that important when the kernel
+  //   does its own page caching?
```

Comment:
From what I could tell, SQLite seems to be pretty sparing with its use of page cache in general. I was trying to get a feel for its limits but, natively, it never seemed to grow past a few MB, even with huge DBs. It's one reason why we introduced our own true LRU page cache, so we could use every bit of memory available.

Reply:
Yeah, SQLite's behavior sounds right if it's assuming the kernel has its own page cache just on the other side of the syscall barrier.